### PR TITLE
HA Bottoms out at 1 DesiredReplica for Utilization and Value metrics

### PR DESCRIPTION
**Pull request by ellistarn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/autoscaler/algorithms/proportional.go [hunk 1]
```diff
@@ -31,15 +31,15 @@ func (a *Proportional) GetDesiredReplicas(metric Metric, replicas int32) int32 {
 	ratio := (metric.Metric.Value / metric.TargetValue)
 	proportional := float64(replicas) * ratio
 	switch metric.TargetType {
-	// Proportional
+	// Proportional, cannot scale to zero
```
Comment (njtran) [inline pkg/autoscaler/algorithms/proportional.go:34]: What about the case where someone wants their HorizontalAutoscaler to scale their resource down all the way, like in the case of a queue? If there's no messages in the queue, I wouldn't want a replica for that, if it was referring to the pods that is.

Comment (ellistarn) [reply pkg/autoscaler/algorithms/proportional.go:34]: This is the AverageValue case. The math doesn't work (divide by zero) if you try to scale to zero using utilization/value metrics.

Comment (njtran) [reply pkg/autoscaler/algorithms/proportional.go:34]: Ah my bad, I misunderstood where I was at in the code.

**Review by njtran — APPROVED** [review]
LGTM

**Pull request merged**